File: ServerFrameWork/EchoServer/Packet/Test_proxy.cpp
#pragma once

#include "Test_proxy.h"

namespace TestC2S
{
	void Proxy::LogInReq( HostID remote, DHNet::RmiContext rmiContext, const String &deviceID, const CDHTime &tmVaue, const std::vector<String> &vecValue )
	{
		auto msg = CMessage::Create();
		msg->SetID( Rmi_LogInReq );
		*msg << deviceID;
		*msg << tmVaue;
		*msg << vecValue;
		RmiSend( remote, rmiContext, Rmi_LogInReq, msg );
	}
}

namespace TestS2C
{
	void Proxy::LogInAck( HostID remote, DHNet::RmiContext rmiContext, const String &deviceID, const CDHTime &tmValue, const std::vector<String> &vecValue )
	{
		auto msg = CMessage::Create();
		msg->SetID( Rmi_LogInAck );
		*msg << deviceID;
		*msg << tmValue;
		*msg << vecValue;
		RmiSend( remote, rmiContext, Rmi_LogInAck, msg );
	}
}

File: ClientFrameWork/Unity3D/Assets/Scripts/Server/Packet/Test_proxy.cs
using System;
using System.Net;
using System.Collections.Generic;


using DHNet;
namespace TestC2S
{
	public class Proxy : DHNet.RmiProxy
	{
		public void LogInReq( RmiContext rmiContext, string deviceID, DateTime tmVaue, List<string> vecValue )
		{
			Message msg = new Message();
			UMessageMarshal.Write( msg, deviceID );
			UMessageMarshal.Write( msg, tmVaue );
			UMessageMarshal.Write( msg, vecValue );
			RmiSend( rmiContext, Common.Rmi_LogInReq, msg );
		}
	}
}

namespace TestS2C
{
	public class Proxy : DHNet.RmiProxy
	{
		public void LogInAck( RmiContext rmiContext, string deviceID, DateTime tmValue, List<string> vecValue )
		{
			Message msg = new Message();
			UMessageMarshal.Write( msg, deviceID );
			UMessageMarshal.Write( msg, tmValue );
			UMessageMarshal.Write( msg, vecValue );
			RmiSend( rmiContext, Common.Rmi_LogInAck, msg );
		}
	}
}

